projects/gloo/api/v1/enterprise/options/ai/ai.proto

```diff
@@ -9,714 +9,738 @@ option (extproto.equal_all) = true;
   message Regex {
+    // Built-in regex patterns for specific types of strings in prompts.
+    // For example, if you specify `CREDIT_CARD`, any credit card numbers
+    // in the request or response are matched.
     enum BuiltIn {
-      // Default REGEX for Social Security Numbers
+      // Default regex matching for Social Security numbers.
       SSN = 0;
-      // Default REGEX for Credit Card Numbers
+      // Default regex matching for credit card numbers.
       CREDIT_CARD = 1;
-      // Default REGEX for Phone Numbers
+      // Default regex matching for phone numbers.
       PHONE_NUMBER = 2;
-      // Default REGEX for Email Addresses
+      // Default regex matching for email addresses.
       EMAIL = 3;
     }
 
+    // Regular expression (regex) matching for prompt guards and data masking.
     message RegexMatch {
-      // The regex pattern to match against the response.
+      // The regex pattern to match against the request or response.
       string pattern = 1;
-      // An optional name for this match which can be used for debugging purposes.
+      // An optional name for this match, which can be used for debugging purposes.
       string name = 2;
     }
-    // A list of Regex patterns to match against the response.
-    // All matches will be masked before being sent back to the client.
-    // matches and builtins are additive.
+    // A list of regex patterns to match against the request or response.
+    // Matches and built-ins are additive.
     repeated RegexMatch matches = 1;
-    // A list of built-in regexes to mask in the response.
-    // matches and builtins are additive.
+    // A list of built-in regex patterns to match against the request or response.
+    // Matches and built-ins are additive.
     repeated BuiltIn builtins = 2;
 
+    // The action to take if a regex pattern is matched in a request or response.
+    // This setting applies only to request matches. Response matches are always masked by default.
     enum Action {
-      // Mask the response if the regex matches
+      // Mask the matched data in the request.
       MASK = 0;
-      // Reject the request if the regex matches
+      // Reject the request if the regex matches content in the request.
       REJECT = 1;
     }
 
-    // The action to take if the regex matches
-    // NOTE: This will only apply to request matches, response matches will always mask
+    // The action to take if a regex pattern is matched in a request or response.
+    // This setting applies only to request matches. Response matches are always masked by default.
     Action action = 3;
   }
 
-  // Webhook settings for prompt guard
+  // Configure a webhook to forward requests or responses to for prompt guarding.
   message Webhook {
     // Host to send the traffic to.
     string host = 1;
     // Port to send the traffic to
     uint32 port = 2;
+    // Describes how to match a given string in HTTP headers. Match is case-sensitive.
```

Comment:
Not a document issue. case-sensitive will never work because HTTP/1 can be camel case but HTTP/2 is always lower case. Also, envoy lower case the header before sending to ext_proc, so the user really need to enter all lower case here for it to work. So, case-sensitive match is incorrect. HTTP header name should be case-insensitive. Should we open an issue to fix this, @EItanya

Reply:
Envoy is all lower case anyway, so not much for us to do there anyway IMO